http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -285,6 +285,22 @@ class HttpWsmDao[F[_]](httpClient: Client[F], config: HttpWsmDaoConfig)(implicit
         AppCreationException(s"could not determine name for resource $resource")
       )
 
+  private def getLandingZoneResourceId(
+    landingZoneResourcesByPurpose: Map[(LandingZoneResourcePurpose, String), List[LandingZoneResource]],
+    resourceType: String,
+    purpose: LandingZoneResourcePurpose,
+    useParent: Boolean
+  ): F[String] =
+    landingZoneResourcesByPurpose
```

Comment:
Could this reuse code in `getLandingZoneResource` and be written as:
```
getLandingZoneResource(landingZoneResourcesByPurpose, resourceType, purpose)
  .flatMap(_.resourceId)
```
?

Reply:
This didn't exactly work, but I was able to rewrite it in a way that uses getLandingZoneResource() instead of querying landingZoneResourcesByPurpose directly.